tests/functional/u/unguarded_next_without_default.py

```diff
@@ -0,0 +1,46 @@
+"""Warnings for using next() without specifying a default value."""
+# pylint: disable=missing-class-docstring, too-few-public-methods, missing-function-docstring
+# pylint: disable=inconsistent-return-statements
+
+next((i for i in (1, 2)), None)
+next(i for i in (1, 2))  # [unguarded-next-without-default]
+var = next(i for i in (1, 2))  # [unguarded-next-without-default]
+
+try:
+    next(i for i in (1, 2))
+except StopIteration:
+    pass
+
+try:
+    next(i for i in (1, 2))  # [unguarded-next-without-default]
+except ValueError:
+    pass
+
+try:
+    next(i for i in (1, 2))
+except (ValueError, StopIteration):
+    pass
+
+try:
+    next(i for i in (1, 2))
+except ValueError:
+    pass
+except StopIteration:
+    pass
+
+redefined_next = next
+redefined_next(i for i in (1, 2))  # [unguarded-next-without-default]
+
+
+class MyClass:
+    def __next__(self):
+        return next(i for i in (1, 2))
```

Comment:
```suggestion
class MyClass:
    def __next__(self):
        return next(i for i in (1, 2))
        
    def expected_to_raise_stop_iteration(self):
        """Custom iterator function that will raise StopIteration by design.

        Raises:
            StopIteration: If the end of the sequence has been reached.
        """
        return next(i for i in (1, 2))        
```
Just a though. Maybe we don't need to check that it's a proper docstring in numpy style or whatever. If there's StopIteration in the docstring we do not raise.